custom_model_runner/datarobot_drum/drum/memory_monitor.py

```diff
@@ -22,10 +22,33 @@ def __init__(self, pid=None, include_childs=True):
         self._include_childs = include_childs
 
     def collect_memory_info(self):
-        current_process = psutil.Process(os.getpid())
-        proc_rss_mb = ByteConv.from_bytes(current_process.memory_info().rss).mbytes
-        for child in current_process.children(recursive=True):
-            proc_rss_mb += ByteConv.from_bytes(child.memory_info().rss).mbytes
+        def get_proc_data(p):
+            return {
+                "pid": p.pid,
+                "cmdline": p.cmdline(),
+                "mem": ByteConv.from_bytes(p.memory_info().rss).mbytes,
+            }
+
+        drum_process = None
+        nginx_rss_mb = 0
+        drum_rss_mb = 0
+
+        for proc in psutil.process_iter():
+            if "drum" in proc.name().lower():
+                if "server" in proc.cmdline():
+                    drum_process = proc
+
+        if drum_process:
```

Comment:
else what?

Reply:
drum_rss=drum_rss_mb + nginx_rss_mb if drum_process else "drum process not found",

Reply:
so drum_process will equal "drum process not found" ? I don't get it.

Reply:
not drum process, but `drum_rss` field in the MemoryInfo, which is only shown in response